contracts/Aavegotchi/facets/ERC1155BuyOrderFacet.sol

```diff
@@ -44,18 +42,15 @@ contract ERC1155BuyOrderFacet is Modifiers {
         uint256 _priceInWei,
         uint256 _quantity,
         uint256 _duration
-    ) external {
+    ) external payable {
         uint256 cost = _quantity * _priceInWei;
         require(cost >= 1e15, "ERC1155BuyOrder: cost should be 0.001 GHST or larger");
 
         address sender = LibMeta.msgSender();
         uint256 category = LibSharedMarketplace.getERC1155Category(_erc1155TokenAddress, _erc1155TokenId);
-        uint256 ghstBalance = IERC20(s.ghstContract).balanceOf(sender);
 
-        // New order
-        // Transfer GHST
-        require(ghstBalance >= cost, "ERC1155BuyOrder: Not enough GHST!");
-        LibERC20.transferFrom(s.ghstContract, sender, address(this), cost);
+        // Transfer ETH
+        require(msg.value == cost, "ERC1155BuyOrder: Not enough GHST sent");
```

Comment:
msg.value might also be > cost...either we change the error message or we allow overdrafts with refunds at the end of the txn

```suggestion
        require(msg.value == cost, "ERC1155BuyOrder:GHST amount mismatch");
```

Reply:
Good catch, I changed this in most other spots. Will update it